pkg/osvscanner/osvscanner.go

```diff
@@ -87,6 +117,49 @@ func scanDir(r *output.Reporter, query *osv.BatchedQuery, dir string, skipGit bo
 	})
 }
 
+type gitIgnoreMatcher struct {
+	matcher  gitignore.Matcher
+	repoPath string
+}
+
+func parseGitIgnores(dir string) (*gitIgnoreMatcher, error) {
+	// We need to parse .gitignore files from the root of the git repo to correctly identify ignored files
+	// Defaults to current directory if dir is not in a repo or some other error
+	// TODO: Won't parse ignores if dir is not in a git repo, and is not under the current directory (e.g ../path/to)
+	// TODO: What is the desired behaviour for non git-repos?
```

Comment:
I think the current behavior of defaulting to current dir as the root is good.

Reply:
Cool - removed that TODO